internal/cmd/operator-sdk/alpha/config3alphato3/convert_config_3-alpha_to_3_test.go

```diff
@@ -0,0 +1,154 @@
+version: 3-alpha
+`
+	basicConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+version: "3"
+`
+)
+
+const (
+	noResourcesConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+version: 3-alpha
+`
+	noResourcesConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+version: "3"
+`
+)
+
+const (
+	complexConfig = `domain: example.com
+layout: go.kubebuilder.io/v3
+projectName: memcached-operator
+resources:
+- crdVersion: v1
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+  webhookVersion: v1
+- crdVersion: v1
+  group: cache
+  kind: MemcachedRS
+  version: v1alpha1
+- group: apps
+  kind: Deployment
+  version: v1
+plugins:
+  manifests.sdk.operatorframework.io/v2: {}
+  scorecard.sdk.operatorframework.io/v2: {}
+version: 3-alpha
+`
+	complexConfigExp = `domain: example.com
+layout: go.kubebuilder.io/v3
+projectName: memcached-operator
+resources:
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  # TODO(user): Update the package path for your API if the below value is incorrect.
+  path: github.com/example/memcached-operator/api/v1alpha1
+  version: v1alpha1
+  webhooks:
+    # TODO(user): Uncomment the below line if this resource's webhook implements a conversion webhook, else delete it.
+    # conversion: true
+    # TODO(user): Uncomment the below line if this resource's webhook implements a defaulting webhook, else delete it.
+    # defaulting: true
+    # TODO(user): Uncomment the below line if this resource's webhook implements a validating webhook, else delete it.
+    # validation: true
+    webhookVersion: v1
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: MemcachedRS
+  # TODO(user): Update the package path for your API if the below value is incorrect.
+  path: github.com/example/memcached-operator/api/v1alpha1
+  version: v1alpha1
+- # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
```

Comment:
Not sure how to handle it but I'm predicting users will not handle the dash appropiately.

Reply:
Yeah this is difficult to handle with templating. If users complain I'd consider trying to fix this.